### DISCUSSION TITLE
Label map

### DISCUSSION THREAD

**Discussion by A-els** [aub-mind/arabert#93] [Q&A] [answered]
Hello, 

I was using the fine-tuning for text classification colab here https://colab.research.google.com/drive/1P9iQHtUH5KUbTVtp8B4-AopZzEEPE0lw?usp=sharing, and upon running the following lines:-

label_map = { v:index for index, v in enumerate(selected_dataset.label_list) }
print(label_map)

I get the following:-
{'neutral': 0, 'negative': 1, 'positive': 2}

After fine-tuning and saving the model my config file has the following:-
"id2label": {
    "0": "LABEL_0",
    "1": "LABEL_1",
    "2": "LABEL_2"
  },

"label2id": {
    "LABEL_0": 0,
    "LABEL_1": 1,
    "LABEL_2": 2
  },

When using the classification pipeline from HuggingFace with:-
classifier = pipeline('sentiment-analysis', model=model, tokenizer=tokenizer, return_all_scores=True)

source code for pipeline - https://huggingface.co/transformers/_modules/transformers/pipelines/text_classification.html#TextClassificationPipeline

The outputs are probabilities for 'LABEL_0' 'LABEL_1' 'LABEL_2'. It makes sense to me that in my case:-
'LABEL_0' = 'neutral'
'LABEL_1' = 'negative'
'LABEL_2' = 'positive' 

And I assume that when I extract the logits from the model using:-
inputs = tokenizer.encode_plus(text_prepped, return_tensors='pt')
outputs = model(**inputs)
outputs['logits']
The logits also correspond to 'neutral' 'negative' 'positive'.

Am I correct in these assumptions or are the labels being mapped in a different way than the way they are presented in the label_map? 

Many thanks.

**Comment by WissamAntoun** [comment, marked as answer]
Yes your assumption is correct.

Btw, before saving the model or before doing `trainer.train()`, you can change the label2id, and id2label maps as follows 

```python
inv_label_map = { v:k for k, v in label_map.items()}
trainer.model.config.label2id = label_map
trainer.model.config.id2label = inv_label_map
```

Which allows the model to directly output the sentiment instead of LABEL_0...

**Reply by A-els** [reply]
Thank you. I edited the config file manually to achieve the same results since I already had it saved but this is a more elegant solution!
